engine/executor/dm/unitholder.go

```diff
@@ -89,105 +68,142 @@ func newUnitHolderImpl(workerType lib.WorkerType, cfg *dmconfig.SubTaskConfig) *
 
 // Init implement UnitHolder.Init
 func (u *unitHolderImpl) Init(ctx context.Context) error {
+	u.processMu.Lock()
```

Comment:
maybe caller can make sure Init is not concurrent with Pause/Resume, so no need to lock here. Otherwise event if we use a lock, Pause/Resume may happen before Init and go wrong.

Also for Close

Reply:
For init it is right, for close the framework needs to be guaranteed, here is to enhance some robustness

Reply:
Maybe we can add unit test to make sure Pause after Close is OK, for example, no panic will happen. (I haven't review unit tests)